pubs/events.py

```diff
@@ -25,17 +25,72 @@ def wrapped_f(*args):
         return wrap
 
 
-class RemoveEvent(Event):
+    # Command events
+
+class PreCommandEvent(Event):
+    description = "Triggered before the command is executed"
+
+class PostCommandEvent(Event):
+    description = "Triggered after the command is executed"
+
+
+    # Paper changes
+
+class PaperChangeEvent(Event):
+    _format = "Unspecified modification of paper {citekey}."
+
     def __init__(self, citekey):
         self.citekey = citekey
 
+    @property
+    def description(self):
+        return self._format.format(citekey=self.citekey)
+
+# Used by repo.push_paper()
+class AddEvent(PaperChangeEvent):
+    _format = "Adds paper {citekey}."
```

Comment:
While I did initially make all the commit messages in present tense, I don't know if that is for the best. Maybe past tense works better.